src/open_inwoner/openzaak/tests/test_clients.py

```diff
@@ -1,28 +1,178 @@
+from unittest import TestCase as PlainTestCase
+
 from django.test import TestCase
 
+import requests
+import requests_mock
 from zgw_consumers.constants import APITypes
-from zgw_consumers.models import Service
 
 from open_inwoner.openzaak.clients import (
+    MultiZgwClientProxy,
+    MultiZgwClientProxyResult,
+    ZgwClientResponse,
     build_catalogi_client,
+    build_catalogi_clients,
     build_documenten_client,
+    build_documenten_clients,
     build_forms_client,
+    build_forms_clients,
     build_zaken_client,
+    build_zaken_clients,
 )
+from open_inwoner.openzaak.exceptions import MultiZgwClientProxyError
 from open_inwoner.openzaak.tests.factories import ZGWApiGroupConfigFactory
 
 
 class ClientFactoryTestCase(TestCase):
     def setUp(self):
-        ZGWApiGroupConfigFactory()
+        self.api_groups = [
+            ZGWApiGroupConfigFactory(name="Default API"),
+            ZGWApiGroupConfigFactory(name="Second API"),
+        ]
 
     def test_originating_service_is_persisted_on_client(self):
-        for factory, api_type in (
-            (build_forms_client, APITypes.orc),
-            (build_zaken_client, APITypes.zrc),
-            (build_documenten_client, APITypes.drc),
-            (build_catalogi_client, APITypes.ztc),
+        for factory, api_type, api_group_field in (
+            (build_forms_client, APITypes.orc, "form_service"),
+            (build_zaken_client, APITypes.zrc, "zrc_service"),
+            (build_documenten_client, APITypes.drc, "drc_service"),
+            (build_catalogi_client, APITypes.ztc, "ztc_service"),
         ):
```

Comment:
Did you consider using subtests for the individual cases (same in the following test)?